haystack/schema.py

```diff
@@ -101,11 +109,13 @@ def __init__(
         allowed_hash_key_attributes = ["content", "content_type", "score", "meta", "embedding"]
 
         if id_hash_keys is not None:
-            if not set(id_hash_keys) <= set(allowed_hash_key_attributes):
+            if not all(key in allowed_hash_key_attributes or key.startswith("meta.") for key in id_hash_keys):
                 raise ValueError(
                     f"You passed custom strings {id_hash_keys} to id_hash_keys which is deprecated. Supply instead a "
-                    f"list of Document's attribute names (like {', '.join(allowed_hash_key_attributes)}). "
-                    "See https://github.com/deepset-ai/haystack/pull/1910 for details)"
+                    f"list of Document's attribute names (like {', '.join(allowed_hash_key_attributes)}) or "
+                    f"a key of meta with a maximum depth of 1 (like meta.url). "
+                    "See https://github.com/deepset-ai/haystack/pull/1910 and "
```

Comment:
please make this link accessible, that means use [link title](link URL)
the same goes for the one below